app/src/org/commcare/utils/StorageUtils.java

```diff
@@ -63,9 +62,10 @@ public static int getNumIncompleteForms() {
             return new FormRecord[0];
         }
 
-        // Order ids so they're submitted to and processed by the server in
-        // the correct order.
-        sortRecordsByDate(ids, storage);
+        if (sortRequired) {
+            // Order ids so they're submitted to and processed by the server in the correct order.
+            sortRecordsBySubmissionOrderingNumber(ids, storage);
```

Comment:
What the reason behind fetching and sorting the ID's first before actually fetching the records?